template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -38,11 +70,67 @@ public List<String> getRoleTypes() {
         return List.of("ATLAS_SERVER");
     }
 
-    @Override
-    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
+    private boolean isAtlasKnoxConfigurationNeeded(TemplatePreparationObject source) {
         return Objects.nonNull(source.getGatewayView())
                 && Objects.nonNull(source.getGatewayView().getExposedServices())
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getAtlasService().getKnoxService());
     }
 
+    private boolean isSDXOptimalizationEnabled(TemplatePreparationObject source) {
+        return source.getStackType().equals(StackType.DATALAKE)
+                && isDatalakeVersionSupported(source.getBlueprintView().getBlueprintText())
+                && entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId());
```

Comment:
Move entitlement check above version check, so we can skip the work needed for version check if it's not useful.